impl/src/bin/cargo-raze.rs

```diff
@@ -81,16 +106,61 @@ fn main() -> Result<()> {
     .and_then(|d| d.deserialize())
     .unwrap_or_else(|e| e.exit());
 
-  // Load settings
-  let manifest_path = PathBuf::from(options
-    .flag_manifest_path
-    .unwrap_or("./Cargo.toml".to_owned())).canonicalize()?;
-  let settings = load_settings(&manifest_path, options.flag_cargo_bin_path.clone())?;
+  options
+}
+
+fn load_raze_settings(options: &Options) -> Result<(Metadata, RazeSettings)> {
+  let metadata = fetch_local_metadata(options)?;
+
+  // Parse settings with that metadata
+  let settings = match load_settings(&metadata) {
+    Ok(settings) => settings,
+    Err(err) => return Err(anyhow!(err.to_string())),
+  };
+
   if options.flag_verbose.unwrap_or(false) {
     println!("Loaded override settings: {:#?}", settings);
   }
 
-  // Fetch metadata
+  Ok((metadata, settings))
+}
+
+fn fetch_local_metadata(options: &Options) -> Result<Metadata> {
+  // Gather basic, offline metadata to parse settings from
+  let fetcher = if let Some(cargo_bin_path) = &options.flag_cargo_bin_path {
+    SettingsMetadataFetcher {
+      cargo_bin_path: PathBuf::from(cargo_bin_path),
+    }
+  } else {
+    SettingsMetadataFetcher::default()
+  };
+
+  let working_directory = if let Some(manifest_path) = &options.flag_manifest_path {
+    let manifest_path = PathBuf::from(manifest_path).canonicalize()?;
+    if !manifest_path.is_file() {
+      return Err(anyhow!("{} is not a file.", manifest_path.display()));
```

Comment:
Optional: prefix this message with "manifest path" so it's immediately clear where the path came from?

(it's probably usually clear, but if the user passed some non-file path here they're probably not real familiar with the software and would benefit from more context)

Reply:
Done